### PR TITLE
fix(client): ensure __runTest is used

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- fix: load framerunner after learner code
- fix: add script id for testing

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

The motivation was to prevent __runTest getting overwritten, but I also like the distinction between test runner and dependencies. I think separating them makes their roles clearer.

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62b46e3a8d4be31be5af793d.md [hunk 1]
```diff
@@ -25,7 +25,10 @@ Your `script` element should come at the end of your `body` element.
 ```js
 const script = document.querySelector('script[data-src$="script.js"]');
 assert.equal(script.previousElementSibling.tagName, "DIV");
-assert.isNull(script.nextElementSibling);
+// When building the test frame, the runner script is always inserted after user
+// code. This means the learner's script should be the penultimate element in
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62b46e3a8d4be31be5af793d.md:29]: Bonus points for using `penultimate`.

**Review by naomi-lgbt — APPROVED** [review]
Doesn't seem to break anything!

**Comment by ShaunSHamilton** [comment]
I do not see anything wrong with this, but I have been trying to understand the workflow.

I see this change really only affects the backend -style challenges. In which case, there is no user input code - all of the code in the test runner is from the Markdown.

So, assuming the backend challenges managed to work in the browser with Camper-code, Campers still could:

```javascript
document.getElementById("fcc-test-runner").remove();
```

Am I misunderstanding this?

**Comment by ojeytonwilliams** [comment]
> I see this change really only affects the backend -style challenges

It affects all challenges that run tests in the browser except JS-only challenges, since they have their own test runner.

> Campers still could:
> document.getElementById("fcc-test-runner").remove();

Not as easily, since the test runner doesn't exist at the time when the user's code is evaluated.

So, in production the order of execution is

1. frame-runner 
2. user code
3. __initTestFrame called
4. tests use __runTest

in this PR it's

1. user code
2. frame-runner
3. __initTestFrame called
4. tests use __runTest

I doubt this is insurmountable, but I like this way structuring the code more and it is somewhat harder to mess with the test suite.

**Review by ShaunSHamilton — APPROVED** [review]
NBWTTTIP 👍

**Pull request merged**